### PR TITLE
Unofficial docs updates

### PR DESCRIPTION
**Pull request by splondike**
Primarily a significant update to the description of .talon files . This hopefully covers everything.

### CODE REVIEW

File: unofficial_talon_docs.md [hunk 1]
```diff
@@ -92,30 +93,28 @@ The following requirements can be set:
...
 
-```
-os: linux
-os: windows
-app: Code
-app: notepad++
-```
-
-This reads: "If OS is either linux or windows, for an app that has a well-known identifier of either Code or notepad++". Code on windows would match, notepad++ on windows would match, and so on; but Code on mac or Sublime on windows would not.
+The next thing to talk about is what happens when we have multiple lines in the context header. Talon lets you build these together as a composite matcher following specific rules. Those rules will be discussed with reference to this unrealistically complicated example:
```
Comment (splondike) [inline unofficial_talon_docs.md:102]: @lunixbochs Correctness review here would be good.

File: unofficial_talon_docs.md [hunk 2]
```diff
@@ -179,19 +190,100 @@ insert lots {user.letter}+:
...
+-
+# Show notification saying the key was pressed and prevent other apps from receiving the key event
+key(f8): app.notify("f8 key pressed")
 
-TODO: Explain how you can override tags and settings in .talon files
+# One or more modifiers can be used with the matcher
+key(ctrl-shift-alt-cmd-super-f8): app.notify("Lots of modifiers and the f8 key pressed")
 
-### Other capabilities
+# Passive mode doesn't work on Windows currently
```
Comment (splondike) [inline unofficial_talon_docs.md:282]: @lunixbochs This is correct right?

Comment (lunixbochs) [reply unofficial_talon_docs.md:282]: :passive and :up are both missing on Windows
the hotkey system also generally has OS-specific quirks

File: unofficial_talon_docs.md [hunk 3]
```diff
@@ -92,30 +93,28 @@ The following requirements can be set:
...
+tag: T1
+tag: T2
+os: O1
+and tag: T3
+tag: T4
+and tag: T5
 ```
 
-This one reads "If either mode is user.talon, or else if mode is command at the same time as code.language is talon".
+If we use T1 etc. to indicate the true/value result of attempting to match the literal string 'T1' against the list of active tags (and similar for os) then this translates to the following boolean expression `(T1 or T2 or (T4 and T5)) and (O1 and T3)`.
```
Comment (lunixbochs) [inline unofficial_talon_docs.md:113]: I believe this is technically correct, but the explanation is sort of confusing and the matches section is in a different order from the resulting boolean expression. Maybe a diagram or overlay would be easier? `app` and `os` are probably more relatable than tag: T1 through T5 as well.

File: unofficial_talon_docs.md [hunk 4]
```diff
@@ -179,19 +190,100 @@ insert lots {user.letter}+:
...
+    # Only a single mathematical operation per line
+    d = 2
+    a = a + d
+    a = a - d
+    a = a * d
+    a = a / d
+    a = a % d
+
+    # or operator is used to deal with optional or alternative command parts, it's not the same
+    # as boolean or. See also the previous code block.
```
Comment (lunixbochs) [inline unofficial_talon_docs.md:215]: not sure this is helpful to our audience, but it's similar to the "null coalescing operator"

File: unofficial_talon_docs.md [hunk 5]
```diff
@@ -179,19 +190,100 @@ insert lots {user.letter}+:
...
+    e = letter or "default"
+
+    # Sleep is a built in function and takes arguments of the (<float>|<integer><suffix>) form.
+    # Float allows specifying (fractions) of a second. The integer+suffix form can be '1m', '5s', '500ms', '1000000us' etc.
+    # Be aware sleeping in this way will prevent Talon from processing voice commands until the
+    # sleep is over
+    sleep(2s)
+
+    # Repeats the previous action or built in command the given number of times,
```
Comment (lunixbochs) [inline unofficial_talon_docs.md:224]: More precisely, I would say "repeats the previous line"

File: unofficial_talon_docs.md [hunk 6]
```diff
@@ -179,19 +190,100 @@ insert lots {user.letter}+:
...
+    # so in this case we'd sleep for a further 4 seconds
+    repeat(2)
+
+    # Allows pressing, holding, and repeating individual key presses. This example holds down
+    # ctrl-alt-a then presses e 5 times, then releases ctrl-alt-a. Look at the basic
+    # customisation page recipes section for a more complete description of the syntax.
+    key("ctrl-alt-a:down e:5 f12 ctrl-alt-a:up")
```
Comment (lunixbochs) [inline unofficial_talon_docs.md:231]: 1. Don't quote the body of key() unless you're doing string interpolation or something that refuses to parse.
2. Should probably have simpler key() example too.

File: unofficial_talon_docs.md [hunk 7]
```diff
@@ -179,19 +190,100 @@ insert lots {user.letter}+:
...
+
+    insert("type in this string")
+    # This is equivalent to insert("type in this string")
+    "type in this string"
```
Comment (lunixbochs) [inline unofficial_talon_docs.md:235]: Bare strings are equivalent to `auto_insert()`, not `insert()`. In general I don't recommend bare strings unless they're the only thing in the script body.

File: unofficial_talon_docs.md [hunk 8]
```diff
@@ -179,19 +190,100 @@ insert lots {user.letter}+:
...
-### Tags, apps, and settings
+```
+title: /my app/
+-
+# Show notification saying the key was pressed and prevent other apps from receiving the key event
+key(f8): app.notify("f8 key pressed")
 
-TODO: Explain how you can override tags and settings in .talon files
+# One or more modifiers can be used with the matcher
+key(ctrl-shift-alt-cmd-super-f8): app.notify("Lots of modifiers and the f8 key pressed")
```
Comment (lunixbochs) [inline unofficial_talon_docs.md:280]: cmd and super are the same

File: unofficial_talon_docs.md [hunk 9]
```diff
@@ -471,51 +563,41 @@ TODO: Describe what modes are, what they're good for, and how to set them up
 
 ### Apps
 
-Talon can activate a context based on which application is active.  It's not unlikely that important apps are part of several .talon files.  Because one and the same app may need to be identified in several different ways (based on platform or app version), Talon allows to register well-known apps, and specify the detailed logic of how to match an app only once.  In all the other places, only the well-known name needs to be used.
-
+Talon can activate a context based on which application is active.  It's not unlikely that important apps are used in several .talon files. It is convenient to reuse any matcher logic that is needed to identify such applications.
```
Comment (lunixbochs) [inline unofficial_talon_docs.md:566]: This seems potentially less clear than the original. The important part is that this example allows you to create a well-known name for an app, not how the Talon scope system works.

File: unofficial_talon_docs.md [hunk 10]
```diff
@@ -36,7 +36,8 @@ os: windows
 os: linux
 app: Slack
 app: Teams
-# Anything above this (single!) dash is part of the context header.
+# Anything above this dash is part of the context header. One or more dashes is permitted, but a
+# single dash is customary.
```
Comment (lunixbochs) [inline unofficial_talon_docs.md:40]: I'm planning on requiring a single dash, so probably don't make this change.

**Pull request merged**